docs/adrs/0009-nimbus-web-sdk-architecture.md

```diff
@@ -0,0 +1,97 @@
+
+## Context and Problem Statement
+
+This is part of a body of work necessary to support the use of Nimbus within web applications. 
+The current Nimbus SDK is written in such a way that it supports client-oriented experimentation — experiments are downloaded, evaluated, and stored on the client, and a feature store is exposed with the experiment branches applied. 
+In order to support web experimentation, the Nimbus SDK will need to be updated to be stateless, to support a more statically defined set of helper methods, and to have additional support for Python.
+
+Ultimately, the problem we're trying to solve can be boiled down to one question — how can we update the Nimbus SDK to support web applications while continuing to support the existing clients?
+
+## Decision Drivers
+
+* Architecture decisions for Nimbus on the web[<sup>[1.i]</sup>](#links)
+  * The core logic of the existing SDK will be re-used for the Nimbus web service (Cirrus)
+  * The SDK needs to support Python
+  * The SDK needs to be stateless
+
+## Considered Options
+
+* Separate `nimbus-core`, `nimbus-sdk`, and `cirrus-sdk` libraries
+* Cargo features, one library
+* Cargo features, separate library
+* Expose a stateless API alongside the stateful API
+
+## Decision Outcome
+
+[justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)].
+
+### Positive Consequences
+
+* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+* …
+
+### Negative Consequences
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* …
+
+## Pros and Cons of the Options
+
+### Separate `nimbus-core`, `nimbus-sdk`, and `cirrus-sdk` libraries
+
+For this option, we would take the core Nimbus business logic currently in the `nimbus-sdk` library and move it to a new `nimbus-core` library. 
+The `nimbus-sdk` and `cirrus-sdk` libraries would depend on the `nimbus-core` library, extending the functionality for their various use cases. 
+
+* Good, because separate libraries allow for strong separation of concerns.
+* Bad, because defining and using types becomes more difficult.
+  * Traits will be needed to extend functionality in the Nimbus Core library. While dynamic typing with traits is supported, it is one of the more complex features of Rust.
+* Bad, because it would result in a significant amount of code churn.
+
+### Cargo features, one library
+
+Cargo features allow for conditional compilation of libraries and library dependencies. 
+Native Rust attributes are applied to sections of Rust code to conditionally compile structs, methods, variables, and more.
+
+For more information, check out the Cargo Features reference[<sup>[1.ii]</sup>](#links) available in the Rust Cargo book.
+
+* Good, because the use of feature configurations is familiar to Rust developers.
+* Good, because all code relating to Nimbus business logic remains in one place.
```

Comment:
Sort of? In this case that's going to specifically be just the core Nimbus business logic, because all logic for statefulness and Kotlin/Swift will end up being in the `nimbus-sdk` crate.